napari/util/misc.py

```diff
@@ -58,6 +59,135 @@ def is_rgb(shape):
         return False
 
 
+def is_pyramid(data):
+    """If data is a list of arrays of decreaing size.
+    """
+    if isinstance(data, list):
+        size = [np.prod(d.shape) for d in data]
+        return np.all(size[:-1] > size[1:])
+    else:
+        return False
+
+
+def should_be_pyramid(shape):
+    """Check if any data axes needs to be pyramided
+
+    Parameters
+    ----------
+    shape : tuple of int
+        Shape of data to be tested
+
+    Returns
+    -------
+    pyr_axes : tuple of bool
+        If any axis is above pyramiding threshold.
+    """
+    return np.log2(shape) > 13
```

Comment:
This function appears to be untested: testing `if should_be_pyramid((45, 435))` will raise an error. Should be `np.any(np.log2(shape) > 13)`. At any rate, `max_size` or something should be a parameter. Doesn't vispy allow up to 16k x 16k, ie log2(shape) == 14?

Reply:
Oh, I see, just saw its use later. Then change the doc to read, "True wherever an axis exceeds the pyramid threshold." And the type returned is an array of bool.

Finally, shouldn't there also be a check for the total size? ie 8k x 8k x 8k will give "should be pyramided" False everywhere, but vispy won't like it?

Reply:
if we do 2D rendering then 8k x 8k x 8k will always be fine - but yes if we go to 3D mode it will choke. I'm not sure how to think about 3D mode

Reply:
I think the cut off for 3D rendering might be more like 1k x 1k x 1k, and I don't want to autopyramid everything that's so big as people might not ever want to look at it in 3D. I could do a downsample on the vispy side just before sending to the graphics card so we don't crash, but I can do that in a new PR. Thoughts @jni?

Reply:
Unclear, let’s punt on this. My intuition is to add a displayed_dims keyword argument that changes the computation, but it wouldn’t be fun to hook that up. We can rely on users for now.